pkg/controller/machine/machine.go

```diff
@@ -327,21 +327,21 @@ func (c *Controller) syncHandler(key string) error {
 	machine := listerMachine.DeepCopy()
 
 	// step 1: check if the machine can be processed by this controller.
-	// set the annotation "machine.k8s.io/controller": my-controller
+	// set the label "machine.k8s.io/controller": my-controller
 	// and the flag --name=my-controller to make only this controller process a node
-	machineControllerName := machine.Annotations[controllerNameAnnotationKey]
+	machineControllerName := machine.Labels[controllerNameLabelKey]
```

Comment:
The main idea behind using a label instead of an annotation is to be able to use a label selector within the informer factories so we do not have to build an exclusion step within the actual business logic